api/src/main/java/uk/gov/dvsa/mot/app/MotrRequestHandler.java

```diff
@@ -0,0 +1,199 @@
+    private static final Logger logger = Logger.getLogger(MotrRequestHandler.class);
+
+    private MotrReadService motrReadService;
+    private HgvVehicleProvider hgvVehicleProvider;
+    private String awsRequestId;
+
+    public MotrRequestHandler() {
+        super();
+    }
+
+    public MotrRequestHandler(boolean injectSelf) {
+        super(injectSelf);
+    }
+
+    @Inject
+    public void setHgvVehicleProvider(HgvVehicleProvider hgvVehicleProvider) {
+        this.hgvVehicleProvider = hgvVehicleProvider;
+    }
+
+    @Inject
+    public void setMotrReadService(MotrReadService motrReadService) {
+        this.motrReadService = motrReadService;
+    }
+
+    @GET
+    @Path("motr/v2/search/registration/{registration}")
+    @Produces("application/json")
+    public Response getVehicle(@PathParam("registration") String registration,
+                               ContainerRequestContext requestContext) throws TradeException {
+        try {
+
+            ApiGatewayRequestContext context = (ApiGatewayRequestContext) requestContext.getProperty(
+                    RequestReader.API_GATEWAY_CONTEXT_PROPERTY);
+            if (context != null) {
+                awsRequestId = context.getRequestId();
+            }
+
+            logger.info(String.format("Entering MotrRequestHandler, awsRequestId: %s", awsRequestId));
+
+            Optional<MotrResponse> motVehiclesOptional = getMotVehicle(registration);
+
+            if (motVehiclesOptional.isPresent()) {
+
+                return Response.ok(buildMotResponse(motVehiclesOptional.get())).build();
+            }
+
+            Optional<MotrResponse> dvlaVehicleOptional = getDvlaVehicle(registration);
+
+            if (dvlaVehicleOptional.isPresent()) {
+                MotrResponse dvlaVehicle = dvlaVehicleOptional.get();
+
+                Optional<Vehicle> hgvPsvVehicleOptional = getHgvPsvVehicle(registration);
+
+                if (!hgvPsvVehicleOptional.isPresent()) {
+                    logger.info("No HGV/PSV vehicle retrieved");
+
+                    if (dvlaVehicle.getMotTestExpiryDate() != null) {
+                        return Response.ok(buildMotResponse(dvlaVehicle)).build();
+                    }
+
+                    throw new InvalidResourceException("No HGV/PSV vehicle retrieved for registration " + registration,
+                            awsRequestId);
+                }
+
+                HgvPsvVehicle hgvPsvVehicle = buildHgvPsvResponse(hgvPsvVehicleOptional.get(), dvlaVehicle);
+
+                return Response.ok(hgvPsvVehicle).build();
+            } else {
+                logger.error("No MOT Test or DVLA vehicle found for registration.");
+                throw new InvalidResourceException("No MOT Test or DVLA vehicle found for registration " + registration,
+                        awsRequestId);
+            }
+        } catch (TradeException e) {
+            // no need to log these errors, just throw them back
```

Comment:
what will happen with this exception when rethrown ?

Reply:
TradeException constructors ensure it has a status code specified. From what I can see some exceptions inherit it, for example:
InvalidResourceException - TradeException with HTTP:404
BadRequestException - TradeException with HTTP:400

Based on what kind of TradeException is caught here we will have different error responses